blog/nse_portfolio_sorts/index.qmd

```diff
@@ -0,0 +1,931 @@
+  mutate(year = year + 1) |> 
+  rename_with(.cols = at, ~ paste0(.x, "_lag"))
+
+# Compute asset growth
+compustat <- compustat |> 
+  left_join(compustat_lag, by = c("gvkey", "year")) |> 
+  mutate(sv_ag = (at - at_lag) / at_lag)
+
+# Compute filters
+compustat <- compustat |> 
+  mutate(filter_be = coalesce(seq, ceq + pstk, at - lt) + coalesce(txditc, txdb + itcb, 0) - coalesce(pstkrv, pstkl, pstk, 0),
+         filter_price = prcc_f,
+         filter_earnings = ib)
+
+# Select required variables
+compustat <- compustat |> 
+  select(gvkey, month, datadate, starts_with("filter_"), starts_with("sv_")) |> 
+  drop_na()
+```
+
+For the CRSP data, we also construct a variable that we will need below. We compute the stock age filter as the time in years between the stock's first appearance in CRSP and the current month. To do this reliably, we again leverage `group_by()`'s power. We also lag the filter by one month to avoid inducing a look-ahead bias before merging it back to our main stock market data. Following our main data construction, market capitalization is already in the CRSP database.
+
+```{r}
+crsp_monthly_filter <- crsp_monthly |>
+  group_by(permno) |>
+  arrange(month) |>
+  mutate(filter_stock_age = as.numeric(difftime(month, min(month), units = "days"))/365,
+         month = month %m+% months(1)) |> 
+  ungroup() |> 
+  select(permno, month, filter_stock_age)
+
+crsp_monthly <- crsp_monthly |> 
+  select(permno, gvkey, month, industry, exchange, mktcap, mktcap_lag, ret_excess) |> 
+  left_join(crsp_monthly_filter, by = c("permno", "month")) |> 
+  drop_na()
+```
+
+At the moment, we only have panels of stock returns and characteristics. These panels still need to be matched together yet, because this also constitutes a decision. Hence, let us move to discuss these decisions.
+
+# The decision nodes
+
+In WWW, we identify 14 methodological choices that must be made to estimate a premium from portfolio sorts. We split these into decisions on the sample construction and the portfolio construction. The table below illustrates the choices, and for further reference, you can refer to WWW for details on these nodes.
+
+Node | Choices
+:------|:------
+Size restriction | none, NYSE 5%, NYSE 20%
+Financials | include, exclude
+Utilities | include, exclude
+Pos. book equity | include, exclude
+Pos. earnings | include, exclude
+Stock-age restriction | none, >2 years
+Price restriction | none, >\$1, >\$5 
+Sorting variable lag | 3 months, 6 months, Fama-French
+Rebalancing | monthly, annually
+Breakpoint quantiles main | 5, 10
+Double sort | single, double dependent, double independent
+Breakpoint quantiles secondary | 2, 5
+Breakpoint exchanges | NYSE, all
+Weighting scheme | equal-weighting, value-weighting
+
+In principle, there are more decisions to be made. However, this set of 14 choices appears in published, peer-reviewed articles and covers different aspects. If you think another choice is essential, I want to hear about it.
```

Comment:
"I want to hear about it." klingt unfreundlich, "please reach out" wäre besser

Reply:
Done in Commit https://github.com/tidy-finance/website/pull/34/commits/a1850e2b8819856a37184cac34149c2cdb841e56.